isisd/isis_sr.c

```diff
@@ -627,6 +627,71 @@ static int sr_local_block_release_label(struct sr_local_block *srlb,
 	return 0;
 }
 
+static bool sr_adj_same_subnet_ipv4(struct in_addr ipv4,
+				    struct isis_circuit *circuit)
+{
+	struct listnode *node;
+	struct prefix_ipv4 *ipv4_adj, *ipv4_circuit;
+	bool is_same = false;
+
+	ipv4_adj = prefix_ipv4_new();
+	if (!ipv4_adj)
+		return false;
+
+	for (ALL_LIST_ELEMENTS_RO(circuit->ip_addrs, node, ipv4_circuit)) {
+		ipv4_adj->prefix = ipv4;
+		ipv4_adj->prefixlen = ipv4_circuit->prefixlen;
+		if (!prefix_cmp((struct prefix *)ipv4_adj,
+				(struct prefix *)ipv4_circuit)) {
+			is_same = true;
+			break;
+		}
+	}
+
+	prefix_ipv4_free(&ipv4_adj);
+
+	return is_same;
+}
+
+static bool sr_adj_same_subnet_ipv6(struct in6_addr *ipv6,
+				    struct isis_circuit *circuit)
+{
+	struct listnode *node;
+	struct prefix_ipv6 *ipv6_adj, *ipv6_circuit;
+	bool is_same = false;
+
+	ipv6_adj = prefix_ipv6_new();
```

Comment:
again no need to allocate.  Just use the stack here.

Reply:
just apply the same criticisms from the v4 version of htis function to the v6

Reply:
ok